app/lib/discourse_automation/scripts/scheduled_explorer_query.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+include HasSanitizableFields
+
+DiscourseAutomation::Scriptable::SCHEDULED_EXPLORER_QUERY = "scheduled_explorer_query"
+
+DiscourseAutomation::Scriptable.add(DiscourseAutomation::Scriptable::SCHEDULED_EXPLORER_QUERY) do
+  queries = []
+  if SiteSetting.data_explorer_enabled
+    queries =
+      DataExplorer::Query.where(hidden: false).map { |q| { id: q.id, translated_name: q.name } }
+  end
+
+  field :recipients, component: :email_group_user, required: true
+  field :query_id, component: :choices, required: true, extra: { content: queries }
+  field :query_params, component: :"key-value", accepts_placeholders: true
+
+  version 1
+  triggerables [:recurring]
+
+  script do |context, fields, automation|
```

Comment:
I know most of them are not artchitectured like this ATM. But what I would recommend now, is to extract all script content into some class. That way you can have a cleaner API, clear method names and it would be easier to test, as you would mostly test behavior and not automation.

Reply:
Thanks, I will look into this and see if it can be cleaned up.